src/parser/parsers/changelogs/parse_changelogs.py

```diff
@@ -1,37 +1,27 @@
-import sys
 import os
-from os import listdir
-from os.path import isfile, join
 import datetime
+from .fetch_changelogs import ChangelogFetcher
 
-# bring utils module in scope
-sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
 import utils.json_utils as json_utils
 
-from .constants import OUTPUT_DIR
-
 
 class ChangelogParser:
-    def __init__(self):
+    def __init__(self, output_dir):
         self.CHANGELOGS_DIR = os.path.join(os.path.dirname(__file__), '../raw-changelogs/')
-        self.OUTPUT_DIR = OUTPUT_DIR
+        self.OUTPUT_DIR = output_dir
         self.OUTPUT_CHANGELOGS = self.OUTPUT_DIR + '/changelogs'
         self.resources = self._get_resources()
+        self.changelogs_by_date = {}
 
-    def run_all(self):
-        changelogs_by_date = {}
-        files = [f for f in listdir(self.CHANGELOGS_DIR) if isfile(join(self.CHANGELOGS_DIR, f))]
-        for file in files:
-            date = file.replace('.txt', '')
-            changelog = self.run(date)
-            changelogs_by_date[date] = changelog
-
+    def run_all(self, txt=True, rss=True):
+        log_fetch = ChangelogFetcher(self.OUTPUT_DIR) 
+        for key, val in log_fetch.fetch_changelogs().items():
```

Comment:
`key, val` is too generic, I think `date, changelog` fits

Reply:
renamed!